hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -1248,6 +1333,14 @@ update env ledger st ev = case (st, ev) of
           onOpenChainCloseTx openState newChainState closedSnapshotNumber contestationDeadline
       | otherwise ->
           Error NotOurHead{ourHeadId, otherHeadId = headId}
+  (Open openState@OpenState{headId = ourHeadId}, ClientInput (SideLoadSnapshot confirmedSnapshot)) ->
+    let otherHeadId =
```

Comment:
You could use `getSnapshot` function here and then just extract the `headId` from the `Snapshot`